packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -189,6 +190,21 @@ export const DataSourceHttpSettings: React.FC<HttpSettingsProps> = props => {
             />
           </div>
 
+          {config?.sigV4AuthEnabled && (
+            <div className="gf-form-inline">
+              <Switch
+                label="SigV4 auth"
+                labelClass="width-13"
+                checked={dataSourceConfig.jsonData.sigV4Auth || false}
+                onChange={event => {
+                  onSettingsChange({
+                    jsonData: { ...dataSourceConfig.jsonData, sigV4Auth: event!.currentTarget.checked },
+                  });
+                }}
+              />
+            </div>
+          )}
```

Comment:
Wonder if instead of passing the entire config object which is used only for retrieving the `sigV4AuthEnabled` flag we should just pass the `sigV4AuthEnabled` property? I guess there might be other auth options in the future, but do we plan to add any short term?

Reply:
Great point @dprokop. It would be unlikely so that's a good change to make I reckon. Thanks for pointing that out 😄